File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        int a = 1, b = 1;
        for (int k=2; k<=n; k++){
            int c = a+b;
            a = b;
            b = c;
        }
        return b;
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        s_prev = 1 # ways to get 0
        s_now  = 1 # ways to get 1
        for k in range(2,n+1):
            s_next = s_prev + s_now # ways to get k
            s_prev = s_now
            s_now = s_next
        return s_now